server/src/main/java/org/elasticsearch/index/query/SearchExecutionContext.java

```diff
@@ -323,6 +325,15 @@ public MappedFieldType getFieldType(String name) {
         return failIfFieldMappingNotFound(name, fieldType(name));
     }
 
+    /**
+     * Returns the registered mapped field types.
+     */
+    public Collection<MappedFieldType> getFieldTypes() {
```

Comment:
Given that this is used only in parent join, I was chatting to Alan and wondering if we can replace this with a method that is more targeted to the specific needs of the parent-join scenario. For instance, allow to retrieve field type of a specific type or something along those lines. I am worried that this is introduces another way to retrieve mapped field types while really in 99% of the cases getFieldType is what consumers should use.